### PR TITLE
FIX: Give Optionals a default value

### LINKED ISSUE #814: Make `num_columns` and `num_rows` actually optional

### LINKED ISSUE #815: Ensure all `Optional` fields have a default
We have seen at least one field other field that does not provide a default value which will still make it a required field.

### PR DESCRIPTION
**Pull request by mferrera**
Resolves #814 
Resolves #815 

I still have a bit uncertainty about some fields. In some cases we give optional lists a default value of an empty list, in some cases it's `null`.

A few instances I added the `Field` or sorted the keyword arguments to it to make the default show first.

I'm working on a test that will run through all the models and fail if an optional doesn't provide a default value... but figured I can put this up before that is working.

### CODE REVIEW

File: src/fmu/dataio/_model/fields.py [hunk 1]
```diff
@@ -103,7 +103,9 @@ class File(BaseModel):
     )
     """The path of a file relative to the case root."""
 
-    checksum_md5: Optional[str] = Field(examples=["kjhsdfvsdlfk23knerknvk23"])
+    checksum_md5: Optional[str] = Field(
+        default=None, examples=["kjhsdfvsdlfk23knerknvk23"]
+    )
```
Comment (mferrera) [inline src/fmu/dataio/_model/fields.py:108]: A test requires that validation cannot occur without checksum, still listed as Optional. Is this a backward compatibility thing patch-handled in Sumo?

File: src/fmu/dataio/_model/specification.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ class SurfaceSpecification(RowColumn):
 class PointSpecification(BaseModel):
     """Specifies relevant values describing an xyz points object."""
 
-    attributes: Optional[List[str]]
+    attributes: Optional[List[str]] = Field(default_factory=list)
```
Comment (mferrera) [inline src/fmu/dataio/_model/specification.py:55]: This needs a closer look?

Comment (tnatt) [reply src/fmu/dataio/_model/specification.py:55]: Yes default=None might be better here

Comment (tnatt) [reply src/fmu/dataio/_model/specification.py:55]: And most likely a pattern we have elsewhere

Comment (mferrera) [reply src/fmu/dataio/_model/specification.py]: Changed to None, maybe I am seeing things 🧐

**Review by tnatt — APPROVED** [review]
Nice 👍 I don't really have a strong opinion on what's best regarding the fields with optional list.. but lean towards having them defaulted to None

**Pull request merged**